rule.py

```diff
@@ -1,100 +1,254 @@
+            ki_da.y = old_pos.x + buffer_y
+        else: 
+            return False 
 
-    def general(self, old_pos, new_pos):
-        pass 
+        if buffer_x == 2: 
+            ki_da.x = old_pos.x + buffer_x - 1
+        elif buffer_x == -2:
+            ki_da.x = old_pos.x + buffer_x + 1    
+        elif abs(buffer_x) == 1: 
+            ki_da.x = old_pos.x + buffer_x 
+        else: 
+            return False 
+        
+        return self.board_map_2d[ki_da.y][ki_da.x] == self._IS_EMPTY 
+
+    def is_interfered_elephant(self, old_pos, new_pos):
+        return self.board_map_2d[old_pos.y + int((new_pos.y - old_pos.y)/2)][old_pos.x + int((new_pos.x - old_pos.x)/2)]
+
+    def is_enemy(self, Piece_1, Piece_2):
+        return Piece_1.team != Piece_2.team
+
+    def get_piece(self, pos):
+        return self.board_map_2d[pos.y][pos.x]
+
+    def get_eaten(self, eater, old_pos, new_pos): 
+        self.board_map_2d[new_pos.y][new_pos.x] = eater 
+        self.board_map_2d[old_pos.y][old_pos.x] = self._IS_EMPTY
+
+    def move_allowed(self, old_pos, new_pos): 
+        '''
+        Call functions from moving_ruleset by the names.
+
+        Soldier case is not done. 
+        TO DO: define Piece object type further into smaller categories of soldiers 
+        '''
+        
+        if is_in_board(new_pos): 
+            if not is_interfered(board_map, old_pos, new_pos):
+                if self.get_piece(old_pos).type != 'soldier':
+                    return  getattr(moving_ruleset, self.get_piece(old_pos).type)(old_pos, new_pos) and self.you_shall_not_pass(old_pos, new_pos)
+                elif self.get_piece(old_pos).pass_river: 
+                    if self.is_in_northern_palace(self.get_piece(old_pos).his_pos[0]):
+                        return moving_ruleset.soldier_passed_river_upper_half(old_pos, new_pos)
+                    elif self.is_in_southern_palace(self.get_piece(old_pos).his_pos[0]):
+                        return moving_ruleset.soldier_passed_river_lower_half(old_pos, new_pos)
+                    else: 
+                        raise ValueError('MovRuleSpecial_6: Unknown soldier position with Piece.pass_river = True')
+                else: 
+                    if self.is_in_northern_palace(self.get_piece(old_pos).his_pos[0]):
+                        return moving_ruleset.soldier_upper_half(old_pos, new_pos)
+                    elif self.is_in_southern_palace(self.get_piece(old_pos).his_pos[0]):
+                        return moving_ruleset.soldier_lower_half(old_pos, new_pos)
+                    else: 
+                        raise ValueError('MovRuleSpecial_7: Unknown soldier position with Piece.pass_river = False')
+            elif is_enemy(self.get_piece(old_pos), self.get_piece(new_pos)): 
+                #Update the map, the occupant get replaced by the new piece 
+                self.get_eaten(self.get_piece(old_pos), old_pos, new_pos)
+            else: return False 
```

Comment:
In case of interference, there are two types: ally in the new_pos, or a piece blocking the horse or the elephant. if a piece is interfered, you need to check whether it is blocked or not first. It it is blocked, return false. if it is not blocked, then there would a piece at the new_pos, which means now you would need to check whether the piece is an ally or enemy.